osu.Game/Screens/Edit/Setup/SetupScreen.cs

```diff
@@ -104,10 +140,56 @@ private void load(OsuColour colours)
                 }
             };
 
+            audioTrackTextBox.Current.BindValueChanged(audioTrackChanged);
+
             foreach (var item in flow.OfType<LabelledTextBox>())
                 item.OnCommit += onCommit;
         }
 
+        public bool ChangeAudioTrack(string path)
+        {
+            var info = new FileInfo(path);
+
+            if (!info.Exists)
+                return false;
+
+            var beatmapFiles = Beatmap.Value.BeatmapSetInfo.Files;
+
+            // remove the old file
+            var oldFile = beatmapFiles.FirstOrDefault(f => f.Filename == Beatmap.Value.Metadata.AudioFile);
+
+            if (oldFile != null)
+            {
+                beatmapFiles.Remove(oldFile);
+                files.Dereference(oldFile.FileInfo);
+            }
+
+            // add the new file
+            IO.FileInfo osuFileInfo;
+
+            using (var stream = info.OpenRead())
+                osuFileInfo = files.Add(stream);
+
+            beatmapFiles.Add(new BeatmapSetFileInfo
+            {
+                FileInfo = osuFileInfo,
+                Filename = info.Name
+            });
+
+            Beatmap.Value.Metadata.AudioFile = info.Name;
+
+            music.ForceReloadCurrentBeatmap();
+
+            editor.UpdateClockSource();
```

Comment:
`editor` is permitted to be null
```suggestion
            editor?.UpdateClockSource();
```

Reply:
Yeah, it is, but I didn't feel the need to fix this since it's likely never going to be called with a null editor. I guess there's no harm adding the check, though.